components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java

```diff
@@ -167,4 +234,214 @@ private boolean isUserAccessAllowed(String username) {
         }
         return false;
     }
+
+    /**
+     * retrieve the existing application of given name
+     *
+     * @param applicationName application name
+     * @param saasApp         value of IsSaasApp attribute of application.
+     * @return existing Application
+     */
+    private OAuthApplicationInfo getExistingApp(String applicationName, boolean saasApp) {
+
+        OAuthApplicationInfo appToReturn = null;
+        OAuthAdminService oAuthAdminService = new OAuthAdminService();
+        try {
+            OAuthConsumerAppDTO consumerAppDTO = oAuthAdminService.
+                    getOAuthApplicationDataByAppName(applicationName);
+            Map<String, String> valueMap = new HashMap<String, String>();
+            valueMap.put(OAUTH_CLIENT_GRANT, consumerAppDTO.getGrantTypes());
+
+            appToReturn = this.fromAppDTOToApplicationInfo(consumerAppDTO.getOauthConsumerKey(),
+                    consumerAppDTO.getApplicationName(), consumerAppDTO.getCallbackUrl(),
+                    consumerAppDTO.getOauthConsumerSecret(), saasApp, null, valueMap);
+
+        } catch (IdentityOAuthAdminException e) {
+            String errorMsg = "error occur while tryng to get OAuth Application data";
+            log.error(errorMsg,e);
```

Comment:
WE can put the errorMsg string inside the log itself since errorMsg variable is not used in anywhere else.